components/provisioner/internal/operations/stages/provisioning/wait_for_installation.go

```diff
@@ -50,7 +55,12 @@ func (s *WaitForInstallationStep) Run(cluster model.Cluster, operation model.Ope
 		return operations.StageResult{}, fmt.Errorf("error: failed to create kubernetes config from raw: %s", err.Error())
 	}
 
-	installationState, err := s.installationClient.CheckInstallationState(k8sConfig)
+	client, ok := s.installationClients[cluster.KymaConfig.Installer]
+	if !ok {
+		return operations.StageResult{}, fmt.Errorf("error: installation client for installation %s does not exist", cluster.KymaConfig.Installer)
+	}
+
+	installationState, err := client.CheckInstallationState(cluster.ID, k8sConfig)
```

Comment:
When installation reaches the timeout -it is reported in logs, installation operation is recorded in DB in the failed state with `error: timeout while processing operation`. But installer workers are not stopped and the installation process is not interrupted on the cluster.

Reply:
The installer and its method `StartKymaDeployment()` doesn't support any channel which could interrupt the installation process and stop goroutine with installation execution. However, it supports [timeout parameter](https://github.com/kyma-project/hydroform/blob/main/parallel-install/example/example.go#L67) which [stops the installation process](https://github.com/kyma-project/hydroform/blob/main/parallel-install/pkg/deployment/deployment.go#L182) based on the value of this parameter. The solution, in this case, could be set `CancelTimeout`  parameter the same as timeout for `wait_for_installlation`. What do you think about it?

Reply:
After discussion, I added a timeout to the parallel installer equal to timeout for `wait_for_installlation` step.